### PR TITLE
doc: Add documentation for OP-TEE verification

### PR DESCRIPTION
**Pull request by matt2048**
Gives a brief description on how to enable the verification for OP-TEE
using the authentication framework in ATF

Signed-off-by: Matt Ma <matt.ma@linaro.org>

### CODE REVIEW

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: documentation/optee_with_auth_framework.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+## OP-TEE with authentication framework in ATF
+This document gives a brief description on how to enable the verification for OP-TEE using the authentication framework in ATF
+
+---
+## What we should do
+* According to [user-guide.md](https://github.com/ARM-software/arm-trusted-firmware/blob/master/docs/user-guide.md), there is no additional specific build options for the verification of OP-TEE, if we have enabled the authentication framework and specified the BL32 build option when build ATF, the BL32 related certificates will be created automatically by the cert_create tool, and then these certificates will be verified during booting up.
```
Comment (jenswi-linaro) [inline documentation/optee_with_auth_framework.md]: `s/when build/when building/`

**Comment by jenswi-linaro** [comment]
Please abbreviate ARM Trusted Firmware as ARM-TF instead of ATF.

File: documentation/optee_with_auth_framework.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+## OP-TEE with authentication framework in ATF
+This document gives a brief description on how to enable the verification for OP-TEE using the authentication framework in ATF
```
Comment (jenswi-linaro) [inline documentation/optee_with_auth_framework.md]: verification of OP-TEE

File: documentation/optee_with_auth_framework.md (hunk 1 above)
Comment (jenswi-linaro) [inline documentation/optee_with_auth_framework.md]: OP-TEE. If we have...

File: documentation/optee_with_auth_framework.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+---
+## What we should do
+* According to [user-guide.md](https://github.com/ARM-software/arm-trusted-firmware/blob/master/docs/user-guide.md), there is no additional specific build options for the verification of OP-TEE, if we have enabled the authentication framework and specified the BL32 build option when build ATF, the BL32 related certificates will be created automatically by the cert_create tool, and then these certificates will be verified during booting up.
+
+
+* To enable the authentication framework, the following steps should be followed according to [user-guide.md](https://github.com/ARM-software/arm-trusted-firmware/blob/master/docs/user-guide.md), and all the details for these build options are also in [user-guide.md](https://github.com/ARM-software/arm-trusted-firmware/blob/master/docs/user-guide.md), more details about the authentication framework, please see [auth-framework.md](https://github.com/ARM-software/arm-trusted-firmware/blob/master/docs/auth-framework.md) and [trusted-board-boot.md](https://github.com/ARM-software/arm-trusted-firmware/blob/master/docs/trusted-board-boot.md).
```
Comment (jenswi-linaro) [inline documentation/optee_with_auth_framework.md]: `s/, and all the detail/. All the details/`

File: documentation/optee_with_auth_framework.md (hunk 1 above)
Comment (matt2048) [inline documentation/optee_with_auth_framework.md]: Thanks jenswi-linaro, ok, I will fix them

File: documentation/optee_with_auth_framework.md (hunk 2 above)
Comment (matt2048) [inline documentation/optee_with_auth_framework.md]: OK, thanks, I will fix them.

File: documentation/optee_with_auth_framework.md (hunk 3 above)
Comment (matt2048) [inline documentation/optee_with_auth_framework.md]: OK, thanks, will fix

**Comment by jbech-linaro** [comment]
This is ready for merge I think? Just a ping in case @jforissier missed it.

**Comment by jforissier** [comment]
@jbech-linaro indeed, thanks for the reminder. I've rebased and pushed.

**Comment by matt2048** [comment]
@jforissier thanks a lot.

**Pull request closed without merging**